### ISSUE TITLE
Error: map reduce failed

### ISSUE THREAD

**Issue by sergeda** [skratchdot/mongodb-schema#2]
@skratchdot 
I've added schema.js to .mongorc.js but
when run 

> db.product.schema();

 I got :

> Error: map reduce failed:{ "ok" : 0, "errmsg" : "unknown m/r field for sharding: limit" } :_getErrorWithCode@src/mongo/shell/utils.js:25:13

What can be the cause of the problem?

**Comment by skratchdot** [comment]
It's been years since I've looked at this (and I haven't done anything with mongodb professionally for a few years).  It looks like `limit` does not work for sharded map-reduce:
https://stackoverflow.com/questions/9432430/mongodb-unknown-m-r-field-for-sharding-limit

I am always passing a limit to the mapReduce call (in the options here): 
https://github.com/skratchdot/mongodb-schema/blob/master/schema.js#L363

Unless a limit of 0 (or non-number limit) is specified:
https://github.com/skratchdot/mongodb-schema/blob/master/schema.js#L338-L342

Maybe try passing a limit of 0?  I hope to revisit the mongo libs I worked on someday.  I haven't even tested them w/ new versions of mongo
